include/lib/map.h

```diff
@@ -4,24 +4,39 @@
 
 #include "stdlib/stdtypes.h"
 
-/** Data structure for a generic map */
+/** Structures for a generic map. */
+struct entry {
```

Comment:
You should move this into the c file. The only struct that should be in the header file is the top level map struct.